### ISSUE TITLE
Redout: Enhanced Edition (517710)

### ISSUE THREAD

**Issue by supashang** [ValveSoftware/Proton#853]
Game works perfect.
Video: https://www.youtube.com/watch?v=APevkn82dQI
Gist: https://gist.github.com/supashang/6050de4e5ed8854f58d7152521048170
Steam ID: https://store.steampowered.com/app/517710/

![screenshot at 2018-08-28 20-29-47](https://user-images.githubusercontent.com/19254047/44746069-4c67fb00-ab01-11e8-8628-c119e9fbabbd.png)

**Comment by dragonfly-lover** [comment]
I don't believe in what i have seen. In my config redout demo has 10-30 FPS (Nvidia 750ti with 396.52 and ryzen5 1500+), and at some points It falls to 1fps. Can you verify your performance or the demo please?

**Comment by dragonfly-lover** [comment]
finally i managed to find why for some people the game is unplayable.
the reason is that some distros (like ubuntu  bionic) install only the 64bit version of libmpg123-0. the game codec works with the 32 bit, so wine activates a software decoding, slowing down the cpu.
the 32bit libs can be installed by terminal:

`sudo apt install libmpg123-0:i386`

then after restart framerate jumps from 1-10fps to 60-80fps on my system. wow!

@kisak-valve should i have to open a request for ubuntu on launchpad or for proton here?

**Comment by kisak-valve** [comment]
Hello @dragonfly-lover, that issue is being tracked at #957.

**Comment by mimattr** [comment]
Ubuntu 16.04, NVIDIA GTX 970, Proton 3.7-6 Beta:

[Steam System Information](https://gist.github.com/mimattr/b0b7f46366e31e254460b9593ce87e51)

Test | Result
-- | --
Singleplayer | Working as expected
Local Co-op | Working as expected
Online Multiplayer | Partially working^
VR | _NOT TESTED_

Configuration | ...
-- | --
Input | Steam Controller & Xbox 360 Controller^^
Display | 1920x1080
Fullscreen | Yes
Preset | Medium^^^
VSync | Off
API | DX11

^ Unable to connect to online sessions
^^ Connecting Xbox 360 Controller overrides Steam Controller
^^^ Default graphics preset is **Epic**

_NOTE: Testing excludes demo product (519880)_

**Comment by founderio** [comment]
Fwiw:
The demo works, although it seems a lot choppier than the full game and the application bar of Gnome 3 pokes through.

The full game works perfectly, though I have not tested Network or VR.
No changes to default game config, no changes to Proton.

Specs: https://gist.github.com/founderio/3295c24d5608d505fcced5205a32e1fc
Tested with Proton 3.7-6 Beta

**Comment by h8672** [comment]
I get performance hiccups, especially in a career map in volcano area trial called "Blue & Red".
I tried various settings and tested them multiple times as performance varied at times until I could finally got to gold medal. In the end at most I got 0.5 seconds off between minimum and max graphics settings.
I didn't notice difference with nvapi fix from https://github.com/ValveSoftware/Proton/issues/1374 either.
Tested using Ubuntu 18.04, graphics driver NVIDIA 396.54 and Proton 3.7-6.

**Comment by dragonfly-lover** [comment]
Well i've further investigated the problem as i could (not enough skilled to use apitrace). My very good performance happens if i don't select "effects". If they are set performance drops down a lot mailing It unplayable. Looking at GNOME task manager i've seen the game uses only my physical CPU cores, always at much less than full speed.memory usage is under 60% so the problem has to be the GPU. I noted that some circuits are extremely slow but their performance May vary by complexity. One thing stands clearly: in the few seconds at the start were veichles are waiting frames are already dropping. So it's not a problem of GPU load, probably, but i think more a problem of shaders cache. My suspect is that dxvk layer requires much more than 128mb shaders cache and the system hangs even when the scene is not moving for these reason.

**Comment by h8672** [comment]
Commented PROTON_USE_NVAPI from my proton user settings and performance issue went away...

**Comment by the-moonwitch** [comment]
Running the game in SteamVR mode with the Valve Index connected freezes my entire pc to the point of requiring a reboot.

[steam-517710.log](https://github.com/ValveSoftware/Proton/files/3349981/steam-517710.log)

**Comment by AwesamLinux** [comment]
**System Information**
- GPU: RX VEGA 56
- GPU Driver: 19.1.0+aco+git1565347502-34dd1ddde6f-1bionic1 (Valves PPA)
- Kernel version: 5.2.8-050208-lowlatency
- Distro version: Linux Mint 19.2 Tina (64 bit) (Cinnamon)
- Proton version: 4.11-2

**Retest Compatibility Report**

There appears to be a **regression** with Proton 4.11-2. Game crashes at the loading screen when trying to start a race. Tried both with and without ACO.

With Proton 4.11-1 the game used to work, and I also tested that it still works with older Proton 4.2-9.

[steam-517710-proton-4.2-9.log](https://github.com/ValveSoftware/Proton/files/3489058/steam-517710-proton-4.2-9.log)
[steam-517710-proton-4.11-2-aco.log](https://github.com/ValveSoftware/Proton/files/3489059/steam-517710-proton-4.11-2-aco.log)
[steam-517710-proton-4.11-2-llvm.log](https://github.com/ValveSoftware/Proton/files/3489060/steam-517710-proton-4.11-2-llvm.log)
[system_info.txt](https://github.com/ValveSoftware/Proton/files/3489062/system_info.txt)

**Reproduction**

- Load the game
- Select career mode
- Try to start a race on the very first track

**Comment by dragonfly-lover** [comment]
i can confirm regression also for my configuration

**Comment by AwesamLinux** [comment]
**System Information**
- GPU: RX VEGA 56
- GPU Driver: 19.1.0+aco+git1566739320-db73bde35cc-1bionic1 (Valves PPA)
- Kernel version: 5.2.10-050210-lowlatency
- Distro version: Linux Mint 19.2 Tina (64 bit) (Cinnamon)
- Proton version: 4.11-3

**Retest Compatibility Report**

With Proton version 4.11-3 the game works perfectly again.

[steam-517710-proton-4.11-3.log](https://github.com/ValveSoftware/Proton/files/3551867/steam-517710-proton-4.11-3.log)
[system_info.txt](https://github.com/ValveSoftware/Proton/files/3551868/system_info.txt)

**Comment by mimattr** [comment]
> Running the game in SteamVR mode with the Valve Index connected freezes my entire pc to the point of requiring a reboot.

@slavfox Do you still have this issue? It appears to work with **Proton 5.13-2** and **SteamVR 1.5.10** though there are a few visual quirks (e.g. incorrect menu position, flat planes obscuring left eye while loading levels)

**Comment by oblitum** [comment]
On latest Proton (6.3-5) this game didn't load for me due to a weird Unread Engine 4 error: It was unable to find the "APEXFramework_x64.dll" from PhysX and gave a fatal error. I end up realizing it was due to filesystem case sensitivity, so I went to `/mnt/data/Steam/steamapps/common/Redout/Engine/Binaries/ThirdParty/PhysX/Win64/VS2015` and `mv ApexFramework_x64.dll APEXFramework_x64.dll`, which fixed the problem and the game then worked flawlessly.

[The same thing happened with the "GRIP: Combat Racing" game](https://github.com/ValveSoftware/Proton/issues/2050#issuecomment-888404161), which is based on the same version of the Unread Engine.

**Comment by rathel** [comment]
Hello, I'm getting this error when I try to run this game, I'm on Gentoo, which I'm probably missing some USE flags, but this is the only game so far I've tried.

Kernel: 5.17.7-gentoo-x86_64
Proton Version: ALL
![2022-05-12_22-05](https://user-images.githubusercontent.com/2610014/168209291-4c06329f-f194-4988-ac93-3d679420730b.png)

Proton Log:
https://gist.github.com/rathel/7d93c27f961625025689e8e50becaa08
Steam System Information:
https://gist.github.com/rathel/6d4bed0608e1a8ab6bf3ccdaa16f8288

**Comment by nded03** [comment]
The developer recently updated the game to a newer UE4 version and broke Proton compatibility. You can roll back to the older working version by going to the game's page clicking the gear icon then Properties->Betas and entering the beta access code "redoutvredition" without quotes.

**Comment by ishitatsuyuki** [comment]
Looks like the developer forgot to add the correct VC Redist dependency.

In the meanwhile, you can solve this by running `protontricks -c 'wine ~/".steam/steam/steamapps/common/Steamworks Shared/_CommonRedist/vcredist/2019/VC_redist.x64.exe"' 517710`.

(Do not use winetricks vcrun2019, it sucks.)

**Comment by d10sfan** [comment]
This game is listed as steam deck verified but just shows up a black screen.

**Comment by waspennator** [comment]
Hdr doesn't work on the oled deck, I tried enabling it after starting the game in dx12 mode and it ends up just visually freezing the screen until you turn it off.

[steam-517710.log](https://github.com/ValveSoftware/Proton/files/13506115/steam-517710.log)

**Comment by waspennator** [comment]
Same result whether I try running it with "PROTON_ENABLE_AMD_AGS=1 %command%" or "PROTON_ENABLE_NVAPI=1 %command%", I do see the text get whiter for a bit when i turn off hdr in dx12 mode, so i dunno what's going on.
